### ISSUE TITLE
2nd step signup form not using form html tag

### ISSUE THREAD

**Issue by NicholasChoong** [codersforcauses/poops#247]
**Is your feature request related to a problem? Please describe.**
The 2nd step component is not using the form tag so `enter` key does not work and there might be some accessibility issues as `div` tags were used to create the component.

**Describe the solution you'd like**
- Change the `div` tag to `form` tag
- Use `react hook form` for the data validation
